### PR TITLE
Add trait for raw text representation

### PR DESCRIPTION
**Pull request by aringenbach**
Add a new trait for raw text representation and implement it in node/document. 
It also updates the helper testing for empty list items.

### CODE REVIEW

File: crates/wysiwyg/src/tests/test_formatting.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ fn selecting_and_bolding_multiple_times() {
         &model.state.dom.to_string(),
         "<strong>aa</strong>bb<strong>cc</strong>"
     );
+    assert_eq!(&model.state.dom.to_raw_text(), "aabbcc");
```
Comment (andybalaam) [inline crates/wysiwyg/src/tests/test_formatting.rs:32]: This would be better as a separate test. Would also be worth checking e.g. an empty Dom, some nested tags, some empty tags etc.

File: crates/wysiwyg/src/tests/test_lists.rs [hunk 2]
```diff
@@ -32,6 +32,7 @@ fn test_ordered_list() {
     assert_eq!(tx(&model), "<ol><li>abcd</li><li>\u{200b}|</li></ol>");
     replace_text(&mut model, "efgh");
     assert_eq!(tx(&model), "<ol><li>abcd</li><li>\u{200b}efgh|</li></ol>");
+    assert_eq!(&model.state.dom.to_raw_text(), "abcd\u{200b}efgh");
```
Comment (andybalaam) [inline crates/wysiwyg/src/tests/test_lists.rs:35]: Again, separate tests would be better, maybe even in a different test file.

**Review by andybalaam — CHANGES_REQUESTED** [review]
Looks great - just requesting some more tests.

**Review by andybalaam — APPROVED** [review]
Nice thanks!
